src/common/meta/src/kv_backend/postgres.rs

```diff
@@ -542,19 +650,108 @@ impl KvBackend for PgStore {
             }
         }
     }
+
+    async fn execute_txn_op(
+        &self,
+        query_executor: &PgQueryExecutor<'_>,
+        op: TxnOp,
+    ) -> Result<TxnOpResponse> {
+        match op {
+            TxnOp::Put(key, value) => {
+                let res = self
+                    .put_with_query_executor(
+                        query_executor,
+                        PutRequest {
+                            key,
+                            value,
+                            prev_kv: false,
+                        },
+                    )
+                    .await?;
+                Ok(TxnOpResponse::ResponsePut(res))
+            }
+            TxnOp::Get(key) => {
+                let res = self
+                    .range_with_query_executor(
+                        query_executor,
+                        RangeRequest {
+                            key,
+                            range_end: vec![],
+                            limit: 1,
+                            keys_only: false,
+                        },
+                    )
+                    .await?;
+                Ok(TxnOpResponse::ResponseGet(res))
+            }
+            TxnOp::Delete(key) => {
+                let res = self
+                    .delete_range_with_query_executor(
+                        query_executor,
+                        DeleteRangeRequest {
+                            key,
+                            range_end: vec![],
+                            prev_kv: false,
+                        },
+                    )
+                    .await?;
+                Ok(TxnOpResponse::ResponseDelete(res))
+            }
+        }
+    }
 }
 
 #[async_trait::async_trait]
 impl TxnService for PgStore {
     type Error = Error;
 
-    async fn txn(&self, _txn: KvTxn) -> Result<KvTxnResponse> {
-        // TODO: implement txn for pg kv backend.
-        unimplemented!()
+    async fn txn(&self, txn: KvTxn) -> Result<KvTxnResponse> {
+        let _timer = METRIC_META_TXN_REQUEST
+            .with_label_values(&["etcd", "txn"])
+            .start_timer();
+
+        let mut client = self.get_client().await?;
+        let pg_txn = PgQueryExecutor::Transaction(client.transaction().await.context(
+            PostgresTransactionSnafu {
+                operation: "start".to_string(),
+            },
+        )?);
+        let mut success = true;
+        if txn.c_when {
+            for cmp in txn.req.compare {
+                let value = self
+                    .point_get_with_query_executor(&pg_txn, &cmp.key)
```

Comment:
Could this cause a performance bottleneck? I think `batch get` would be more appropriate.